osu.Game/Overlays/Music/PlaylistItem.cs

```diff
@@ -114,24 +117,62 @@ private void recreateText()
             });
         }
 
-        protected override bool OnHover(Framework.Input.InputState state)
+        protected override bool OnHover(InputState state)
         {
             handle.FadeIn(fade_duration);
 
             return base.OnHover(state);
         }
 
-        protected override void OnHoverLost(Framework.Input.InputState state)
+        protected override void OnHoverLost(InputState state)
         {
             handle.FadeOut(fade_duration);
         }
 
-        protected override bool OnClick(Framework.Input.InputState state)
+        protected override bool OnClick(InputState state)
         {
             OnSelect?.Invoke(BeatmapSetInfo);
             return true;
         }
 
+        protected override bool OnDragStart(InputState state)
+        {
+            return true;
+        }
+
+        // Maybe render some ghost text
+        protected override bool OnDrag(InputState state)
+        {
+            return true;
+        }
+
+        protected override bool OnDragEnd(InputState state)

```

Comment:
How about moving this code to OnDrag instead?

Reply:
It only needs to be computed once. From what I saw OnDrag is constantly being invoked while the object is being dragged.

Reply:
That's exactly what I'd expect though - instantaneous feedback of some sort. Otherwise you need to add the "ghost text" or whatever you had in mind.

I personally had in mind the item popping out and providing a blank space, however I would've been happy with the effect this causes if it's recomputed in OnDrag.

The computation itself is in-expensive and has the correct src == dst check.

Reply:
Now that I think about it, it is inexpensive. When I was initially working on it, I was thinking that the computation is not very cheap since you have to shift all affected tracks. However, realistically you only move it by 1 track at a time so at most it's affecting two tracks. I will make the change.